x-pack/plugins/observability_solution/logs_shared/server/deprecations/log_sources_setting.ts

```diff
@@ -7,67 +7,94 @@
 import { DeprecationsDetails } from '@kbn/core-deprecations-common';
 import { GetDeprecationsContext } from '@kbn/core-deprecations-server';
 import { i18n } from '@kbn/i18n';
+import pMap from 'p-map';
 import { defaultLogViewId } from '../../common/log_views';
 import { MIGRATE_LOG_VIEW_SETTINGS_URL } from '../../common/http_api/deprecations';
 import { logSourcesKibanaAdvancedSettingRT } from '../../common';
 import { LogsSharedPluginStartServicesAccessor } from '../types';
 
-export const getLogSourcesSettingDeprecationInfo = async ({
-  getStartServices,
-  context,
-}: {
+export interface LogSourcesSettingDeprecationParams {
   context: GetDeprecationsContext;
   getStartServices: LogsSharedPluginStartServicesAccessor;
-}): Promise<DeprecationsDetails[]> => {
-  const [_, pluginStartDeps, pluginStart] = await getStartServices();
-  const logSourcesService =
-    pluginStartDeps.logsDataAccess.services.logSourcesServiceFactory.getLogSourcesService(
-      context.savedObjectsClient
+}
+
+export const getLogSourcesSettingDeprecationInfo = async (
+  params: LogSourcesSettingDeprecationParams
+): Promise<DeprecationsDetails[]> => {
+  const allAvailableSpaces = await params.context.savedObjectsClient.getSearchableNamespaces(['*']);
```

Comment:
Requires end users to actively enable Log sources in each space.
We need that documented.